scripts/machine-launch-script.sh

```diff
@@ -0,0 +1,305 @@
+    echo "  % $0 --prefix ~/conda --env my_custom_env"
+    echo "     Install into $HOME/conda and add install to ~/.bashrc"
+    echo "  % $0 --prefix \${CONDA_EXE%/bin/conda} --env my_custom_env"
+    echo "     Create my_custom_env in existing conda install"
+    echo "     NOTES:"
+    echo "       * CONDA_EXE is set in your environment when you activate a conda env"
+    echo "       * my_custom_env will not be activated by default at login see /etc/profile.d/conda.sh & ~/.bashrc"
+}
+
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+        --help)
+            usage
+            exit 1
+            ;;
+        --prefix)
+            shift
+            CONDA_INSTALL_PREFIX="$1"
+            shift
+            ;;
+        --env)
+            shift
+            CONDA_ENV_NAME="$1"
+            shift
+            if [[ "$CONDA_ENV_NAME" == "base" ]]; then
+                echo "::ERROR:: best practice is to install into a named environment, not base. Aborting."
+                exit 1
+            fi
+            ;;
+        --dry-run)
+            shift
+            DRY_RUN_OPTION="--dry-run"
+            DRY_RUN_ECHO=(echo "Would Run:")
+            ;;
+        --reinstall-conda)
+            shift
+            REINSTALL_CONDA=1
+            ;;
+        *)
+            echo "Invalid Argument: $1"
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+if [[ $REINSTALL_CONDA -eq 1 && -n "$DRY_RUN_OPTION" ]]; then
+    echo "::ERROR:: --dry-run and --reinstall-conda are mutually exclusive.  Pick one or the other."
+fi
+
+set -ex
+set -o pipefail
+
+{
+
+    # uname options are not portable so do what https://www.gnu.org/software/coreutils/faq/coreutils-faq.html#uname-is-system-specific
+    # suggests and iteratively probe the system type
+    if ! type uname >&/dev/null; then
+	echo "::ERROR:: need 'uname' command available to determine if we support this sytem"
+	exit 1
+    fi
+
+    if [[ "$(uname)" != "Linux" ]]; then
+        echo "::ERROR:: $0 only supports 'Linux' not '$(uname)'"
+	exit 1
+    fi
+
+    if [[ "$(uname -mo)" != "x86_64 GNU/Linux" ]]; then
+        echo "::ERROR:: $0 only supports 'x86_64 GNU/Linux' not '$(uname -io)'"
+        exit 1
+    fi
+
+    if [[ ! -r /etc/os-release ]]; then
+        echo "::ERROR:: $0 depends on /etc/os-release for distro-specific setup and it doesn't exist here"
+        exit 1
+    fi
+
+    OS_FLAVOR=$(grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+    OS_VERSION=$(grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+
+    echo "machine launch script started" > machine-launchstatus
+    chmod ugo+r machine-launchstatus
+
+    # platform-specific setup
+    case "$OS_FLAVOR" in
+        ubuntu)
+            ;;
+        centos)
+            ;;
+        *)
+            echo "::ERROR:: Unknown OS flavor '$OS_FLAVOR'. Unable to do platform-specific setup."
+            exit 1
+            ;;
```

Comment:
What's the point of this? To say we only support ubunctu and centos?
Should the comment say `# detect supported platforms` instead?